sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/CharVarcharUtils.scala

```diff
@@ -222,16 +155,14 @@ object CharVarcharUtils extends Logging {
           classOf[CharVarcharCodegenUtils],
           StringType,
           "charTypeWriteSideCheck",
-          expr :: Literal(length) :: Nil,
-          propagateNull = false)
+          expr :: Literal(length) :: Nil)
```

Comment:
we should set `returnNullable` to `false`.

Reply:
good point~